### DISCUSSION TITLE
Unable to run fastlane on Apple Silicon (MacBook Air M1)

### DISCUSSION THREAD

**Discussion by alakoring** [fastlane/fastlane#17788] [Q&A]
I’d been happily using fastlane on macOS 10.14. I just switched to an M1 device running Big Sur and no dice.

I’ve tried different installation techniques

```
 % sudo bundle update
sudo: unable to execute /usr/local/bin/bundle: No such file or directory

```
or

```
% brew install fastlane
Error: No Homebrew ruby 2.6.3_2 available for arm64 processors!
```
or

```% gem install fastlane
…
Done installing documentation for CFPropertyList, jwt, multi_json, aws-eventstream, aws-sdk-core, fastlane after 18 seconds
6 gems installed
% fastlane
fastlane: Command not found.
```

Clearly it’s installed

```
% which fastlane
/usr/local/bin/fastlane
```
but it looks like it is trying to use the wrong Ruby — the first line

`#!/System/Library/Frameworks/Ruby.framework/Versions/2.4/usr/bin/ruby
`
refers to a nonexistent file

```
% /System/Library/Frameworks/Ruby.framework/Versions/2.4/usr/bin/ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.4/usr/bin/ruby: Command not found.
```

This is weird because

```
% which ruby
/usr/local/opt/ruby@2.4/bin/ruby
```
I tried `sudo gem uninstall fastlane` and reinstalled it, but the results are the same.

I haven’t seen any documentation about this. What am I missing?

**Comment by alakoring** [comment]
I tried manually editing `/usr/local/bin/fastlane` to refer to `/usr/local/opt/ruby@2.4/bin/ruby` and it starts, but then I get

`[18:34:05]: ▸ sh: /usr/local/bin/xcpretty: /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby: bad interpreter: No such file or directory`

so there is still some weird mismatch going on.

**Reply by sami79031** [reply]
I have the same issue with the new M1 (Mac mini).

**Reply by dinsen** [reply]
@sami79031  did you restore the Mac mini from Time Machine or something?
Try see this: https://github.com/fastlane/fastlane/issues/17820

**Reply by sami79031** [reply]
@Dinsen It's a new Mac mini I just bought.

**Reply by sami79031** [reply]
Updating ruby, reinstalling fastlane fixed it.

**Reply by lucasmpaim** [reply]
I'm with the same problem, witch ruby version did you use @sami79031 ?

**Reply by sami79031** [reply]
> I'm with the same problem, witch ruby version did you use @sami79031 ?

I'm using `ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]`

**Reply by lucasmpaim** [reply]
I'm using fastlane with m1 macbook pro, for work I have to use the https://rvm.io/ and install the ruby 2.7 version (for work all plugins that I need to use).

```
ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [arm64-darwin20]
```

**Reply by jesphinpt** [reply]
I am using fastlane in m1 mac machine.

But after installing the fastlane successfully, it shows the error as below if I try to check the version.(/Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:69:in `parse_global_options': ambiguous option: --v (OptionParser::AmbiguousOption)
)

```
which ruby
/usr/bin/ruby

```
```
 which fastlane
/usr/local/bin/fastlane
```

```
fastlane --v
[✔] 🚀 
[12:35:23]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
Traceback (most recent call last):
	8: from /usr/local/bin/fastlane:23:in `<main>'
	7: from /usr/local/bin/fastlane:23:in `load'
	6: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/bin/fastlane:23:in `<top (required)>'
	5: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	4: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	3: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
	2: from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	1: from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:91:in `run!'
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:69:in `parse_global_options': ambiguous option: --v (OptionParser::AmbiguousOption)

fastlane -v

fastlane installation at path:
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.197.0/bin/fastlane
-----------------------------
[✔] 🚀 
fastlane 2.197.0
```

If I try to install rvm it shows 

```
rvm install 3.0.2

Warning: your HOMEBREW_PREFIX is set to /usr/local but HOMEBREW_CELLAR is set
to /opt/homebrew/Cellar. Your current HOMEBREW_CELLAR location will stop
you being able to use all the binary packages (bottles) Homebrew provides. We
recommend you move your HOMEBREW_CELLAR to /usr/local/Cellar which will get you
access to all bottles."
Installing requirements for osx.
Updating system - please wait
Failed to update Homebrew, follow instructions at

    https://docs.brew.sh/Common-Issues

and make sure `brew update` works before continuing.
Error running 'requirements_osx_brew_update_system ruby-3.0.2',
please read /Users/name/.rvm/log/1635836879_ruby-3.0.2/update_system.log
Requirements installation failed with status: 1.
```

brew update shows

```
Warning: your HOMEBREW_PREFIX is set to /usr/local but HOMEBREW_CELLAR is set
to /opt/homebrew/Cellar. Your current HOMEBREW_CELLAR location will stop
you being able to use all the binary packages (bottles) Homebrew provides. We
recommend you move your HOMEBREW_CELLAR to /usr/local/Cellar which will get you
access to all bottles."
mkdir: /usr/local/var/homebrew/locks: Permission denied
Error: Can't create update lock in /usr/local/var/homebrew/locks!
Fix permissions by running:
  sudo chown -R $(whoami) /usr/local/var/homebrew

```

homebrew is available in the /opt/homebrew in my new machine.

**Comment by JavyMB** [comment]
HI @alakoring  I had a similar issue in a Mac Mini M1 and after a lot of reading I found that **fastlane** is not working properly with HomeBrew in M1.🧑🏻‍🔧
Try uninstalling it from home brew 
`brew uninstall fastlane`
and  install it as a Gem command: 👀
`sudo gem install fastlane -NV`

Again... this worked for me but Im not sure it will be a definitive answer. 

Here's the Article I used https://narlei.com/development/apple-m1-install-fastlane/
